File: Codewars/8kyu/8kyu_fake_binary/src/main.py
import unittest

def fake_bin(x):
    # return ''.join('0' if 5 > int(ch) else '1' for ch in x)
    return ''.join('0' if '5' > ch else '1' for ch in x)


class Test(unittest.TestCase):
    def test_1(self):
        tests = [
            # [expected, input]
            ["01011110001100111", "45385593107843568"],
            ["101000111101101", "509321967506747"],
            ["011011110000101010000011011", "366058562030849490134388085"],
            ["01111100", "15889923"],
            ["100111001111", "800857237867"],
        ]

        for exp, inp in tests:
            self.assertEqual(fake_bin(inp), exp)


if __name__ == '__main__':
    unittest.main()

File: Codewars/8kyu/8kyu_fake_binary/src/main.js
function mysolution(x) {
    return x.split('').map(x => x >= 5? 1: 0).join('')
}

function othersolution(x) {
    return x.replace(/\d/g, d => d < 5? 0: 1)
}

fakeBin = othersolution

console.log(fakeBin('45385593107843568') === '01011110001100111');
console.log(fakeBin('509321967506747') === '101000111101101'); 
console.log(fakeBin('366058562030849490134388085') === '011011110000101010000011011'); 